### PR TITLE
filter for active, overide N returned, cap 100

### PR DESCRIPTION
**Pull request by daverhub**

### CODE REVIEW

File: lib/pinball.js [hunk 1]
```diff
@@ -530,9 +533,15 @@ class Pinball {
             }
           }
           if( aArray.length !== 0){
+            let nActiveCount = aArray.filter( (a) =>{return (Date.now() - a.val) < nearbyMinutesThreshold}).length
```
Comment (daverhub) [inline lib/pinball.js]: count number of incidents active in last 20 minues

File: lib/pinball.js [hunk 2]
```diff
@@ -530,9 +533,15 @@ class Pinball {
...
+            nActive = nActiveCount < 100 ? nActiveCount : 100
```
Comment (daverhub) [inline lib/pinball.js]: cap nActive to 100

File: lib/pinball.js [hunk 3]
```diff
@@ -530,9 +533,15 @@ class Pinball {
...
           if( aArray.length !== 0){
+            let nActiveCount = aArray.filter( (a) =>{return (Date.now() - a.val) < nearbyMinutesThreshold}).length
+            nActive = nActiveCount < 100 ? nActiveCount : 100 
             aArray.sort( (a,b) => a.val - b.val );
           }
-
+          
+          if (nActive > N) {
+            // return all active 
+            N = nActive
```
Comment (daverhub) [inline lib/pinball.js]: overide N returned with NActive

**Comment by victusfate** [comment]
spoke with @daverhub , he tested locally ready to merge

**Comment by victusfate** [comment]
@daverhub ok updated the code, not really loving it.

**Comment by victusfate** [comment]
I'd say a good 80-90% of the cpu time is sitting in the sorts now based on the results array dimensions (now often 100), and other requirements (level based search)

**Comment by smmcl** [comment]
This is deprecated and will not be merged.

**Pull request closed without merging**